components/Button/button.test.tsx

```diff
@@ -0,0 +1,60 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Button } from './button';
+
+describe("Button", () => {
+    it("renders the Button Component", () => {
+        render(<Button data-testid="button"/>);
+        const testButton = screen.getByTestId('button');
+        expect(testButton).toBeInTheDocument();
+    });
+
+    it('renders the Button Component with custom className', () => {
+        render(<Button className="custom-class" data-testid="button">Test</Button>);
+        const button = screen.getByTestId('button');
+        expect(button).toHaveClass('custom-class');
+      });
+
+      test.each([
+        ["default", "bg-primary text-primary-foreground shadow"],
+        ["secondary", "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80"],
+        ["destructive", "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90"],
+        ["outline", "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground"],
+        ["ghost", "hover:bg-accent hover:text-accent-foreground"],
+        ["link", "text-primary underline-offset-4 hover:underline"],
+      ])("Renders the Button Component with variant='%s'", (btnVariant, expectedClass) => {
```

Comment:
Refactor (blocking): `btnVariant` is a string and variant expects `"link" | "default" | "secondary" | "destructive" | "outline" | "ghost" | null | undefined`

```suggestion
      test.each([
        ["default", "bg-primary text-primary-foreground shadow"],
        ["secondary", "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80"],
        ["destructive", "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90"],
        ["outline", "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground"],
        ["ghost", "hover:bg-accent hover:text-accent-foreground"],
        ["link", "text-primary underline-offset-4 hover:underline"],
      ] as const)("Renders the Button Component with variant='%s'", (btnVariant, expectedClass) => {
```

![CleanShot 2025-05-19 at 13 19 49](https://github.com/user-attachments/assets/f39fa901-e2f4-4817-8861-16dff079fcdc)